plugins/org.polarsys.capella.scenario.editor.dsl/xtend-gen/org/polarsys/capella/scenario/editor/dsl/helpers/TextualScenarioHelper.java

```diff
@@ -136,4 +126,22 @@ public static boolean isParticipantKeyword(final String keywordValue) {
       "configuration_item", "function");
     return participantsKeywords.contains(keywordValue);
   }
+  
+  public static EList<EObject> getElements(final EObject modelContainer) {
+    if ((modelContainer instanceof Model)) {
+      return ((Model) modelContainer).getElements();
+    }
+    if ((modelContainer instanceof CombinedFragment)) {
+      EList<EObject> elements = ((CombinedFragment) modelContainer).getBlock().getBlockElements();
+      elements.addAll(((CombinedFragment) modelContainer).getOperands());
+      return elements;

```

Comment:
Is ok to return here the Operands?
Elements in grammar are: (Message | Reference | CombinedFragment | StateFragment)

Reply:
in this case, elements are the elements inside a combined fragment, not model elements
maybe the name of the method could be changed - the method returns all elements on the same level :-?

Reply:
You can add a comment on the getElements function explaining the result of return.